### PR TITLE
DCOS-34530: json file mock

### PR DESCRIPTION
**Pull request by nLight**
With `getMockFromJSON` you can turn any JSON to a mock as easy as

```js
{
  id: "my-json-mock",
  request: getMockForJSON({value: 42})
}
```

it will stringify JavaScript Object for you and set "Content-Type: Application/json" header.

Closes DCOS-34530

### CODE REVIEW

File: src/mocks.ts [hunk 1]
```diff
@@ -29,6 +30,23 @@ export const mocks = [];`);
...
 }
 
+export function getMockForJSON(json: string | {}): RequestHandler {
+  if (!json) {
+    throw new Error("JSON argument is missing!");
+  }
+
+  let jsonString = json;
+
+  if (typeof json !== "string") {
```
Comment (juliangieseke) [inline src/mocks.ts]: If I get this correctly, `getMockForJSON("foo")` would result in an invalid output. so if a string is passed, you should try something like `JSON.stringify(JSON.parse(json))`.

File: src/__tests__/mocks-test.ts [hunk 2]
```diff
@@ -70,4 +70,41 @@ describe("mocks", () => {
...
+      }).not.toThrow();
+    });
+
+    it("doesn't throw when valid json string is provided", () => {
+      expect(() => {
+        getMockForJSON("{}");
+      }).not.toThrow();
+    });
+
+    it("returns doesn't throw when json is provided", () => {
```
Comment (juliangieseke) [inline src/__tests__/mocks-test.ts]: thre `returns` doesnt make sense here 🤔

File: src/mocks.ts [hunk 3]
```diff
@@ -29,6 +30,21 @@ export const mocks = [];`);
   return mocks as IMock[];
 }
 
+export function getMockForJSON(json: string | {}): RequestHandler {
```
Comment (DanielMSchmidt) [inline src/mocks.ts]: Just curious, I assume `{}` is a different way of writing `object`, right? It's not "an empty object", is it?

File: src/server.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ import { createProxyServer } from "http-proxy";
 import { IServerConfig } from "./types";
 import { AddressInfo } from "net";
 
+export { getMockForJSON } from "./mocks";
```
Comment (DanielMSchmidt) [inline src/server.ts]: TIL, awesome <3

File: src/mocks.ts (hunk 3 above)
Comment (nLight) [reply src/mocks.ts]: Most probably I'm misusing it. I thought of it as a "plain object" but it's different actually https://blog.mariusschulz.com/2017/02/24/typescript-2-2-the-object-type#object-vs-object-vs maybe we wanna use `object`

Comment (DanielMSchmidt) [reply src/mocks.ts]: I am still learning TS, but I am always using object so I have no knowledge about the other way.

**Comment by nLight** [comment]
@juliangieseke please give it another shot!

**Pull request merged**